src/julia_gc.c

```diff
@@ -37,10 +37,13 @@
 **  Various options controlling special features of the Julia GC code follow
 */
 
-// if DISABLE_BIGVAL_TRACKING is defined, we don't track the location of
-// large bags; this speeds up some things, but may expose bugs in GAP code
-// which incorrectly holds pointers into bags over a GC.
-// #define DISABLE_BIGVAL_TRACKING
+// if SCAN_STACK_FOR_MPTRS_ONLY is defined, stack scanning will only
+// look for references to master pointers, but not bags themselves.
+// This should be safe, as Gasman uses the same mechanism, but it is
+// at least theoretically possible that an optimizing compiler will
+// remove the last master pointer reference before the last reference
+// to the bag itself.
```

Comment:
As discussed several times before: such an optimization would necessarily also affect GASMAN in the very same way! Except perhaps if the Julia GC was running concurrently to the thread GAP is running in; but then we might have a serious problem anyway.

Reply:
To be clear: With that remark I meant to say that I think your comment is misleading, and should be adjusted. Either removed, or qualified.